public/js/ace-1.2.5/worker-latex.js

```diff
@@ -1414,67 +1414,412 @@ var Mirror = require("../worker/mirror").Mirror;
 
 var LatexWorker = exports.LatexWorker = function(sender) {
     Mirror.call(this, sender);
-    this.setTimeout(500);
+    this.setTimeout(250);
 };
 
 oop.inherits(LatexWorker, Mirror);
 
 var Parse = function (text) {
-    var lines = text.split('\n');
-    var state = [], errors = [];
-    var i;
-    var regex = /^\s*\\(begin|end)\{(\w+\*?)\}/; // keep the regex outside the loop for performance
-    for (i = 0; i < lines.length; i++) {
-	var line = lines[i];
-	var result;
-	var error = null;
-	regex.lastIndex = 0;
-	if ((result = regex.exec(line))) {
-	    var type = result[1];
-	    var env = result[2];
-	    if (type == "begin") {
-		state.push({"env":env, "row": i});
-	    } else if (type == "end") {
-		var last_open = state.pop();
-		if (last_open &&  last_open.env == env) {
-		} else if (last_open && last_open.env != env) {
-		    error = {"type":"info", "text": "end " + env + "with begin " + last_open.env, "start_row": last_open.row, "end_row":i };
-		} else if (!last_open) {
-		    error = {"type":"info", "text": "end without begin " + type, "start_row": 0, "end_row": i};
+    var errors = [];
+    var Comments = [];
+    var Tokens = [];
+    var Environments = [];
+    var pos = -1;
+    var SPECIAL = /[\\\{\}\$\&\#\^\_\~\%]/g;
+    var CS = /[^a-zA-Z]/g;
+    var idx = 0;
+    var lineNumber = 0;
+    var linePosition = [];
+    linePosition[0] = 0;
+
+    var TokenError = function (token, message) {
+	var line = token[0], type = token[1], start = token[2], end = token[3], seq = token[4];
```

Comment:
is there a reason the token is an array not object, i.e. token.line, token.type like we do in errors.push